etc/docs/api.proto

```diff
@@ -0,0 +1,63 @@
+service ExecutorService{
+  rpc ExecuteCommand (ExecuteRequest) returns (ExecuteReply) {}
+  rpc StopExecution (ExecutionIdRequest) returns (StopCommandReply) {}
+  rpc ListExecutions (EmptyRequest) returns (ExecutionListReply) {}
+  rpc GetExecution (ExecutionIdRequest) returns (ExecutionInfoReply) {}
+  rpc GetOutStream(ExecutionIdRequest) returns (stream OutEntry) {}
+  rpc GetOut(ExecutionIdRequest) returns (OutEntriesReply) {}
+}
+
+message ExecuteRequest {
+  required string command = 1;
+  repeated string args = 2;
+}
+
+message ExecutionIdRequest {
+  required ExecutionId executionId = 1;
+}
+
+message EmptyRequest {}
+
+message ExecutionId {
+  required string value = 1;
+}
+
+message ExecuteReply {
+  required ExecutionId executionId = 1;
+}
+
+message ExecutionListReply {
+  repeated ExecutionInfoReply executions = 1;
+}
+
+message ExecutionInfoReply {
+  required ExecutionId executionId = 1;
+  enum ExecutionStatus {
+    UNKNOWN = 0;
+    FINISHED = 1;
+    RUNNING = 2;
+    ABORTED = 3;
+    NOT_STARTED = 4;
+  }
+  required ExecutionStatus status = 2;
+  optional int32 exitCode = 3;
+}
+
+message StopCommandReply {
+  required ExecutionId executionId = 1;
+  enum StopStatus {
+    UNKNOWN = 0;
+    STOPPED = 1;
+    KILLED = 2;
+  }
+  required StopStatus status = 2;
+  required int32 exitCode = 3;
+}
+
+message OutEntry {
+  required string line = 1;
```

Comment:
Streaming line by line may be a reasonable assumption but why not make this a bit more flexible by streaming bytes? This way you won't need to worry about commands that e.g. produce large log lines and support non-text output as well.